Update: TERRA Workflow Documentation (#999)
nextstrain/ncov: docs/src/guides/run-ingest-on-terra.rst

**************************

> Run Data Ingest on Terra

Reviewer: The capitalization here makes it seem like "Data Ingest" is some sort of proper noun within Nextstrain, when that's not the case.

```suggestion
Ingest SARS-CoV-2 data on Terra
```

**************************

We have provided two pipelines for importing data into Terra:

* GenBank Ingest - pull a public dataset and send them through our preprocessing scripts.

> * GISAID Ingest - pull a private dataset if a user has their own API key, account, and password.

Reviewer 2: Minor change of words here to be exact about the requirements:

```suggestion
* GISAID Ingest - pull a private dataset if a user has their own API endpoint, username, and password.
```

> The pipelines were mainly motivated to provide access to our data pre-processing scripts. Currently, these are focused on pulling datasets for the ncov workflow and is roughly diagramed below:

Reviewer 3: `diagramed` -> `diagrammed`

.. image:: ../images/terra-ingest.png

> Import ``ingest`` wdl workflow from Dockstore

Reviewer: `ingest` isn't the exact name so I wouldn't stylize it as an inline literal. Also, inline literals in headings are rendered a bit small in the preview. This might be good enough given the existing context:

```suggestion
Import the wdl workflow from Dockstore
```

=============================================

1. `Setup a Terra account <https://terra.bio/>`_

> #. Navigate to Dockstore: `ncov:wdl/genbank_ingest`_ or `ncov:wdl/gisaid_ingest`_ depending on if you wish to pull open (genbank) data or private (and have an gisaid api key) data.

Reviewer: ```suggestion
#. Navigate to Dockstore: `ncov:wdl/genbank_ingest`_ or `ncov:wdl/gisaid_ingest`_ depending on if you wish to pull open (genbank) data or private (and have an GISAID API key) data.
```

Reviewer 2: To be exact, I would update to `GISAID API endpoint`

#. Top right corner, under **Launch with**, click on **Terra**
#. Under "Workflow Name" set a name, such as ``genbank_ingest`` or ``gisaid_ingest``, and select your "Destination Workspace" in the drop down menu.
#. Click button **IMPORT**
#. In your workspace, click on the **WORKFLOWS** tab and verify that the imported workflow is showing a card

.. _`ncov:wdl/genbank_ingest`: https://dockstore.org/workflows/github.com/nextstrain/ncov:wdl/genbank_ingest?tab=info
.. _`ncov:wdl/gisaid_ingest`: https://dockstore.org/workflows/github.com/nextstrain/ncov:wdl/gisaid_ingest?tab=info

Create Terra Variables for GISAID API
=====================================

> If you are pulling GISAID data you must have your own API key. If you are pulling GenBank data (open), click on your imported "genbank_ingest" and skip to step 6.

Reviewer 2: Ditto `API endpoint`

1. Navigate to your workspace on Terra
#. On the **Data** tab, from the left menu click **Workspace Data**
#. Create and fill in values for the following workspace variables:

  +-----------------------------+----------------------------+-----------------------------------------------+
  |Key                          | Value                      | Description                                   |
  +=============================+============================+===============================================+

>   |GISAID_API_ENDPOINT          | url api enpoint value here | Provided by GISAID for your account           |

Reviewer 3: Probably should standardize capitalization throughout. For example:

`url api enpoint value here` -> `URL API endpoint value here`

Note "enpoint" typo, too.

  +-----------------+-------------------------------+-------+----------------------------------------+
  |Nextstrain_WRKFLW|  GISAID_USERNAME_AND_PASSWORD | String| workspace.GISAID_USERNAME_AND_PASSWORD |
  +-----------------+-------------------------------+-------+----------------------------------------+

6. Click on the **OUTPUTS** tab
#. Connect your generated output back to the workspace data, but filling in values:

  +-----------------+------------------+-------+----------------------------------+
  |Task name        | Variable	       | Type  |   Attribute                      |
  +=================+==================+=======+==================================+
  |Nextstrain_WRKFLW|  sequences_fasta | File  | workspace.gisaid_sequences_fasta |
  +-----------------+------------------+-------+----------------------------------+
  |Nextstrain_WRKFLW|  metadata_tsv    | File  | workspace.gisaid_metadata_tsv    |
  +-----------------+------------------+-------+----------------------------------+
  |Nextstrain_WRKFLW|  nextclade_tsv   | File  | workspace.gisaid_nextclade_tsv   |
  +-----------------+------------------+-------+----------------------------------+

>   If you are pulling GenBank data, use something like ``workspace.genbank_sequences_fasta`` instead.

Author: Completely agree about adding tabs for readability! I just didn't think of it, thanks!

>   If you are pulling GenBank data, use something like ``workspace.genbank_sequences_fasta`` instead.

Reviewer: Yes that sounds fine! Or, given that there is not much text outside of the tabs in your current work, another option would be to just have two separate pages:

- Terra: Ingest SARS-CoV-2 data from GISAID
- Terra: Ingest SARS-CoV-2 data from GenBank

>   If you are pulling GenBank data, use something like ``workspace.genbank_sequences_fasta`` instead.

Author: Agree! :D And I can make that happen with an easy change
